src/Entity/Content.php

```diff
@@ -445,16 +445,32 @@ public function removeTaxonomy(Taxonomy $taxonomy): self
     /**
      * Generic getter for a record fields. Will return the field with $name.
      *
+     * If $name is not found, throw an exception if it's invoked from code, and
+     * return null if invoked from within a template. In templates we need to be
+     * more lenient, in order to do things like `{% if record.foo %}..{% endif %}
+     *
+     * Note: We can not rely on `{% if record.foo is defined %}`, because it
+     * always returns `true` for object properties.
+     * See: https://craftcms.stackexchange.com/questions/2116/twig-is-defined-always-returning-true
```

Comment:
Have you tried using http://php.net/manual/en/language.oop5.overloading.php#object.isset instead?

Reply:
I hadn't before, but I did try now.. When doing something like this: 

```
    public function __isset($name)
    {
        try {
            $this->getField($name);
            return true;
        } catch (\InvalidArgumentException $e) {
            return false;
        }
    }
```

The behaviour remains unchanged. If I break it down, it seems that the following happens with this twig snippet: 

```twig
    {% if record.foo is defined %}
        It "record.foo" should be defined, we get here
        {#{{ dump(record.foo) }}#}
    {% endif %}
```

When `record.foo` is parsed in the `{% if %}`, it correctly calls my overridden `__isset()`. My overridden isset _correctly_ returns `false` to twig. Then the `is defined` test is parsed, and it still *incorreclty* returns `true`, causing the part within the `{% if %}` to be output. :-/

Reply:
Upon further investigation, overriding `__isset()` is no good, for two reasons: 

 1) It's ignored in Twig's `defined` test. Even if I do this, the results get ignored:

```php
    public function __isset(string $name)
    {
        return null;
    }
```

2) The problem is in Twig. Overriding it, will have side effects for when calling it from PHP code, making it more complex and potentially buggy. 

So, let's stick with the general direction of the workaround I've made above. I've added an e2e test to make sure it works as expected. 

As an additional benefit, it'll make cleaner, more expressive templates for implementors, because we can do `{% if record.foo %}…`, even in strict mode. If you really need to test if the _field_ exists, you can do `{% if record.foo is null %}`.

Reply:
How about changing __call to __get, will that work with __isset ?

Reply:
> If you really need to test if the field exists, you can do {% if record.foo is null %}.

Looks like a nightmare if you want to check if field has the value of null...

Reply:
If you just want to check if field has been filled, you can use `record.foo|default`

Reply:
Rebased this PR to master, and it's quite a bit simpler as a result. 

> How about changing __call to __get, will that work with __isset ? 

Nope, it doesn't. :-)

> Looks like a nightmare if you want to check if field has the value of null...

Thinking some more on this.. the correct syntax in Twig would be: `{% if record.foo is null %}`. But, the common fields _never_ return `null`. Only '``' or `false`. 

I can't really think of a usecase where you'd need to distinguish in Twig between a field that doesn't exist, _or_ a field that does exist but has value `null`. In these cases, you're writing the template so you most likely know which case you'll be dealing with.

Reply:
If you really need to test if the field exists, you can do `{% if record.hasField('foo') %}`.

^ that's the correct way :)

And I agree that usecases for it would be rather rare.